pkg/virtualization/core/virtualsystem/virtualsystemsettingdata.go

```diff
@@ -251,6 +252,21 @@ func (vm *VirtualSystemSettingData) GetVirtualHardDisks() (col disk.VirtualHardD
 	return
 }
 
+func (vm *VirtualSystemSettingData) GetVirtualDvdDrives() (col drive.DvdDriveCollection, err error) {
+	resourceType := fmt.Sprintf("%d", int32(v2.ResourceAllocationSettingData_ResourceType_DVD_drive))
+	query := query.NewWmiQuery("Cim_ResourceAllocationSettingData", "ResourceType", resourceType)
+	rasdcollection, err := instance.GetWmiInstancesFromHost(vm.GetWmiHost(), string(constant.Virtualization), query)
+	if err != nil {
+		return
+	}
+
+	col, err = drive.NewDvdDriveCollection(rasdcollection)
+	if err != nil {
+		rasdcollection.Close()
```

Comment:
Where do we call "Close()" for rasdcollection if there is no error here?

Reply:
It is closed in function `GetDvdConfigByIsoPath()` line 780